### ISSUE TITLE
Feature Request: IC2 EU-Reader Upgrade for the Power Fist

### ISSUE THREAD

**Issue by StefanH-AT** [MachineMuse/MachineMusePowersuits#925]
Since there's also a treetap upgrade, I think it makes sense to add the EU-Reader as an upgrade to the Power Fist aswell.

**Comment by eyeonus** [comment]
Does not the OmniProbe module cover this?

**Comment by StefanH-AT** [comment]
I don't have that upgrade availible?

**Comment by eyeonus** [comment]
Okay, lehjr, you're going to have to take this over.

I don't know what version this guy's (gal's) on, I don't know on which version the OmniProbe was added, and I don't know which mod that module depends on.

**Comment by lehjr** [comment]
OmniProbe was added in 1.7.10, carried over from the addons mod. Looking at the code, it hasn't changed much since then, and surprisingly, doesn't look like it eve supported IC2.

That said, this late in 1.12.2, it isn't worth adding new features as I am already tied up with porting (rewriting) for 1.13.2 and it doesn't make sense to spend time on a version that will soon be outdated.

**Comment by StefanH-AT** [comment]
Makes sense. The suggestion still holds though!
I don't know if IC2 is already availible for 1.13.2 but I'd love to sed this feature

**Comment by lehjr** [comment]
That may or may not happen. One of the issues with IC2 is Industrial Craft Classic, which uses the same mod ID but is not compatible code-wise. The other issue is the lack of capabilities for their energy system and the lack of optional interfaces in 1.13.2.

**Comment by lehjr** [comment]
Update: I noticed none of the OmniProbe code was working so I updated all that. It will be fixed in the next release. Unfortunately, due to the way the EU reader is coded, especially with the GUI, it's not something I can replicate.

**Comment by eyeonus** [comment]
Out of curiosity, is the OmniProbe based off TheOneProbe?

While I personally never use IC2 if I can avoid it, I know that TheOneProbe works with that stuff.

**Comment by lehjr** [comment]
No, it's just a wrapper that uses the original mod's meters. The problem with IC2 is that it uses a GUI from onItemFirstUse and getting the itemstack from the player's hand, which in this case is the power fist and won't pass the checks. Even bypassing all that, I still couldn't get it to work. Basically, any time you want to open anther mods GUI, you're going to have issues.

**Comment by eyeonus** [comment]
I don't how much hassle it would be to do, possibly not much if it's using the capability thing, but it might be a good idea to replace the OmniProbe module with a TheOneProbe module instead.

I might look into it if I ever have the time.

https://github.com/McJtyMods/TheOneProbe

**Comment by lehjr** [comment]
Doing an enhanced version of WALIA as a module to me just doesn't seem worth it especially since you can do the same thing without it. 
However, the module itself has probably outlived it's usefulness, it will probably be dropped in 1.13.2.
